cmd/aks-periscope/aks-periscope.go

```diff
@@ -62,9 +73,19 @@ func main() {
 
 	waitgroup.Wait()
 
+	//create map of all possible diagnosers by name
+	allDiagnosersByName := make(map[string]interfaces.Diagnoser)
```

Comment:
💡 Same suggestions here: If we encapsulate this to smaller functs and name this more meaningfully then latter it could be easily testable as well:

Could name this as: `getEnabledDiagnosers` we can assign return of that to the `diagnoser` of this function.

Reply:
refactored